app/controllers/pictures_controller.rb

```diff
@@ -0,0 +1,33 @@
+class PicturesController < ApplicationController
+
+ before_action :authenticate_user!, :except => [:index, :show]
+
+  def index
+    @pictures = Picture.all
+  end
+
+  def new
+    @picture = Picture.new
+  end
+
+  def create
+    @picture = Picture.create(picture_params)
+    redirect_to '/pictures'
+  end
+
+  def picture_params
+    params.require(:picture).permit(:caption, :image)
+  end
+
+  def edit
+    @picture = Picture.find(params[:id])
+  end
+
+  def update
+    @picture = Picture.find(params[:id])
+    @picture.update(picture_params)
```

Comment:
Same thing here - just handle potential errors whenever you're connecting to a DB